src/utils/address.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  checkAddressChecksum,
+  toChecksumAddress,
+  isAddress,
+  isHexStrict,
+} from 'web3-utils';
+
+const getAddressWithoutNetworkPrefix = (address = ''): string => {
+  const hasPrefix = address.includes(':');
+
+  if (!hasPrefix) {
+    return address;
+  }
+
+  const [, ...addressWithoutNetworkPrefix] = address.split(':');
+
+  return addressWithoutNetworkPrefix.join('');
+};
+
+const getNetworkPrefix = (address = ''): string => {
+  const splitAddress = address.split(':');
+  const hasPrefixDefined = splitAddress.length > 1;
+  const [prefix] = splitAddress;
+  return hasPrefixDefined ? prefix : '';
+};
+
+const addNetworkPrefix = (
+  address: string,
+  prefix: string | undefined
+): string => {
+  return !!prefix ? `${prefix}:${address}` : address;
+};
+
+const checksumAddress = (address: string): string => toChecksumAddress(address);
+
+const isChecksumAddress = (address?: string): boolean => {
+  if (address) {
+    return checkAddressChecksum(address);
+  }
+
+  return false;
+};
+
+const isValidAddress = (address?: string): boolean => {
+  if (address) {
+    // `isAddress` do not require the string to start with `0x`
+    // `isHexStrict` ensures the address to start with `0x` aside from being a valid hex string
+    return isHexStrict(address) && isAddress(address);
+  }
+
+  return false;
+};
+
+// Based on https://docs.ens.domains/dapp-developer-guide/resolving-names
+// [...] a correct integration of ENS treats any dot-separated name as a potential ENS name [...]
+const isValidEnsName = (name: string): boolean => name.includes('.');
```

Comment:
This only checks that there's a dot but not any other symbols? `.` is not a valid ens name

Reply:
Good catch!! I will update it to match with `something.somethig`

What is your opinion about this regex?

```
const isENSRegex = new RegExp(/[^\[\]]+\.[^\[\]]/)
```

![Captura de pantalla 2021-11-22 a las 16 23 48](https://user-images.githubusercontent.com/26763673/142888175-5155001c-3345-4059-bd98-c2bcac168b97.png)